### ISSUE TITLE
SMB authentification

### ISSUE THREAD

**Issue by Krzysztof-Lempicki** [apple/cups#5857]
Hi,

CUPS is using **SMB** link like: 

`DeviceURI smb://domain\myAccount:mypassword@server.com/print/PrinterName`

When I change link to: 

```
AuthInfoRequired password
DeviceURI smb://domain\myAccount@server.com/print/PrinterName
```

I can't connect to printer (invalid credentials).
CUPS don't use the password that I provided in password dialog.

How to fix this?

**Comment by michaelrsweet** [comment]
Don't specify a username in the device URI - if the smb backend (*not* part of CUPS) sees user credentials in the URI it will ignore any information supplied in auth-info.

**Comment by grothesque** [comment]
Is there any other way to achieve what @Krzysztof-Lempicki tried to do, i.e. specify the user name but not the password for a SMB printer?  (I've spent a lot of time searching, but in vain so far.)

In my case, the local user name does not match what the printer expects, so each time I print I have to correct it. I know I could specify both in the DeviceURI line, but I do not want to store my password in that file.

I tried setting the option `AuthInfo` to my remote user name, but this does not seem to have any effect.
